make some changes to allow for persistent postgres storage, running charts with db's using tilt, GEAR-527, some updates to default values

paulmurdoch19 opened the pull request.

nss10 committed:
Add cronjob and update audit/values.yaml

nss10 committed:
Update comment

nss10 committed:
Eliminate unwanted comments

george42-ctds committed:
(HP-2279) Update versions in README

george42-ctds committed:
(HP-2279) Add suspend config

nss10 committed:
Update service account for the export job

nss10 committed:
Update service account for job, and set enabled:false as default for the exportjob

nss10 committed:
Update readme with deployment instructions

nss10 committed:
Merge branch 'master' of https://github.com/uc-cdis/gen3-helm into chore/audit_svc_cronjob

nss10 committed:
Update Gen3 chart version

nss10 committed:
Merge branch 'master' of https://github.com/uc-cdis/gen3-helm into chore/audit_svc_cronjob

nss10 committed:
Version bump to fix lint errors

nss10 committed:
Merge branch 'master' into chore/audit_svc_cronjob

jawadqur committed:
Do not allow access to Dockerfile and README.md in nginx

EliseCastle23 committed:
Updating Ssjdispatcher SA Logic (#397)

* updating ssjdispatcher so "create: false" is respected for the ssjdispatcher service account.

* version bump

krishnaa05 committed:
Remove Jenkins Secrets from integration_tests.yaml (#402)

AidanHilt committed:
Override bitnami to legacy repo

AidanHilt committed:
Bump revproxy version

AidanHilt committed:
Bump gen3 chart

AidanHilt committed:
Merge branch 'master' into feat/PPS-1987

jawadqur committed:
Merge pull request #404 from uc-cdis/feat/bitnami-legacy-migration

jawadqur committed:
Merge pull request #400 from uc-cdis/feat/PPS-1987

paulmurdoch19 committed:
Update Helm values and portal configs for new features

Updated container image tags for gearbox, guppy, and portal. Added command and args overrides to amanuensis Helm chart. Enhanced portal configuration with new filter, table, and field mapping options, including survival analysis and unit calculation configs. Modified database setup job template for improved host referencing. Disabled amanuensis and PostgreSQL persistence by default, and adjusted tier access limit.

paulmurdoch19 committed:
Set jwt_public_key.pem secret as required in Fence config

Changed the 'optional' field for the jwt_public_key.pem secret from true to false in pcdc-default-values.yaml, ensuring the key must be present for Fence to function properly.

paulmurdoch19 committed:
Add biospecimen_status field and update external references

Added the 'biospecimen_status' field to the portal configuration in pcdc-default-values.yaml, including it in relevant field lists and table columns. Also updated several subject submitter_id values in pcdc_data/external/external_reference.json. Additionally, removed the 'data - survival' block from the portal configuration.

nss10 committed:
Get audit logs since last successful run, with fallback to last 1 month. Update job SA.

nss10 committed:
Merge branch 'master' of https://github.com/uc-cdis/gen3-helm into chore/audit_svc_cronjob

nss10 committed:
Update gen3 chart version

mfshao committed:
Update portal-service.conf

mfshao committed:
update

george42-ctds committed:
Merge branch 'master' into feat/configure-agg-mds-schedule

george42-ctds committed:
Bump gen3 chart version

george42-ctds committed:
Revert default schedule

jawadqur committed:
Merge pull request #406 from uc-cdis/mfshao-patch-1

Update portal-service.conf

george42-ctds committed:
Merge branch 'master' into feat/configure-agg-mds-schedule

nss10 committed:
Merge branch 'master' into chore/audit_svc_cronjob

nss10 committed:
gen3 chart -- version bump

george42-ctds committed:
Bump chart version

nss10 committed:
Merge pull request #374 from uc-cdis/chore/audit_svc_cronjob

Add cronjob and update audit/values.yaml

george42-ctds committed:
Bump chart version

george42-ctds committed:
Merge branch 'master' into feat/configure-agg-mds-schedule

tianj7 committed:
Migrate DCF redaction and validation jobs (#385)

* add redaction and validation job

george42-ctds committed:
Bump chart version

george42-ctds committed:
Merge branch 'master' into feat/configure-agg-mds-schedule

paulmurdoch19 committed:
Update Helm job and enable services with DB configs

Set imagePullPolicy to IfNotPresent for the cleanup Helm hook job. Enabled multiple services (arborist, amanuensis, fence, peregrine, portal, revproxy, sheepdog, elasticsearch) and added or updated their Postgres password configurations. Increased global tierAccessLimit from 1 to 5 and enabled persistence for PostgreSQL primary.

paulmurdoch19 committed:
Add default postgres passwords for services

Introduced default postgres passwords for arborist, fence, and gearbox services in the configuration file to facilitate local development and testing.

paulmurdoch19 committed:
Add default PostgreSQL configuration values

Introduced default values for PostgreSQL authentication and persistence, including a sample password and storage size, to the gearbox-default-values.yaml file.

kellcuc commented on gearbox-default-values.yaml:
```diff
@@ -55,11 +55,15 @@ global:
 
 arborist:
+  postgres:
+    password: "arborist_thisisaweakpassword"
```
What does this password do?

kellcuc commented on gearbox-default-values.yaml:
```diff
@@ -55,11 +55,15 @@ global:
 
 fence:
+  postgres:
+    password: "fence_thisisaweakpassword"
```
What does this password do?

kellcuc commented on gearbox-default-values.yaml:
```diff
@@ -448,10 +452,12 @@ revproxy:
 
 gearbox:
+  postgres:
+    password: "gearbox_thisisaweakpassword"
```
What does this password do?

kellcuc commented on gearbox-default-values.yaml:
```diff
@@ -480,6 +488,13 @@ gearbox-middleware:
 
+postgresql:
+  auth:
+    postgresPassword: "thisisaterriblepassword"
```
What does this password do?

kellcuc commented on gearbox-default-values.yaml:
```diff
@@ -480,6 +488,13 @@ gearbox-middleware:
     # runAsUser: 1000
     # runAsGroup: 1000
 
+postgresql:
```
Think you need to add 
```
image:
    repository: bitnamilegacy/postgresql
```

kellcuc commented on pcdc-default-values.yaml:
```diff
@@ -57,18 +71,26 @@ global:
 arborist:
+  enabled: true
+  postgres:
+    password: "arborist_thisisaweakpassword"
```
What does this password do?

kellcuc commented on pcdc-default-values.yaml:
```diff
@@ -57,18 +71,26 @@ global:
 amanuensis:
   enabled: true
+  postgres:
+    password: "amanuensis_thisisaweakpassword"
```
What does this password do?

kellcuc commented on pcdc-default-values.yaml:
```diff
@@ -57,18 +71,26 @@ global:
 fence:
+  enabled: true
+  postgres:
+    password: "fence_thisisaweakpassword"
```
What does this password do?

kellcuc commented on pcdc-default-values.yaml:
```diff
@@ -317,15 +447,18 @@ pcdcanalysistools:
 peregrine:
+  enabled: true
+  postgres:
+    password: "peregrine_thisisaweakpassword"
```
What does this password do?

kellcuc commented:
Curious about the addition of the password, what does that add/do? No need to respond to each, probably a single reply would be fine

george42-ctds committed:
Merge pull request #383 from uc-cdis/feat/configure-agg-mds-schedule

Feat/configure agg mds schedule

paulmurdoch19 committed:
Set PostgreSQL image repository to bitnamilegacy/postgresql

Updated the postgresql.image.repository field in both gearbox-default-values.yaml and pcdc-default-values.yaml to use the bitnamilegacy/postgresql image. This ensures consistency in the PostgreSQL image source across both configuration files.

paulmurdoch19 commented:
when you start pcdc/gearbox with persistence on and if no password is provided in the values file then the dbcreds secret randomly generates one and that value is then used as the password for the postgres user that has access to the DB. when you tear down pcdc/gearbox the user with that randomly generated password is save in the persistence volume claim but the dbcreds secret is delete, then when you restart pcdc/gearbox there is no way to access the password. By providing a password in the values file you avoid that. I am also going to add a note that this is only for local dev

paulmurdoch19 committed:
Add dev-only comments to weak password fields

Inserted '#local Dev Only' comments above hardcoded weak passwords in postgres configurations across multiple services in gearbox-default-values.yaml and pcdc-default-values.yaml to clarify these values are for local development use only.

github-actions[bot] committed:
Merge remote-tracking branch 'upstream/master' into sync-cdis-pcdc

paulmurdoch19 committed:
Merge branch 'sync-cdis-pcdc' into updates
